docs/swagger_v3/contracts.spec.yaml

```
@@ -1,4 +1,25 @@
 schemas:
+  ContractSourceTxType:
+    description: Contract source transaction type
+    type: string
+    enum:
+      - ContractCreateTx
+      - GaAttachTx
+      - PayingForTx
+  ContractInternalTx:
+    description: Contract internal transaction
+    type: object
+    anyOf:
```

```suggestion
    oneOf:
```
or it is not working because of missed discriminator?

---

I've manly used anyOf, because you mentioned that it doesn't work in autorest. Should I put oneOf as it should be wherever I've done that?

---

Both `anyOf` and `oneOf` are not working in autorest. Don't worry about autorest I have some workarounds to apply on my side. Do it according to OAS primarly.